app/audit/models/ChargeSummaryAudit.scala

```diff
@@ -49,22 +51,27 @@ case class ChargeSummaryAudit(mtdItUser: MtdItUser[_], docDateDetail: DocumentDe
-    case Some("chargeSummary.paymentAllocations.bcd.incomeTax") => "Income Tax for remaining balance"
-    case Some("chargeSummary.paymentAllocations.bcd.nic2") => "Class 2 National Insurance for remaining balance"
-    case Some("chargeSummary.paymentAllocations.bcd.vcnic2") => "Voluntary Class 2 National Insurance for remaining balance"
-    case Some("chargeSummary.paymentAllocations.bcd.nic4") => "Class 4 National Insurance for remaining balance"
-    case Some("chargeSummary.paymentAllocations.bcd.sl") => "Student Loans for remaining balance"
-    case Some("chargeSummary.paymentAllocations.bcd.cgt") => "Capital Gains Tax for remaining balance"
-    case Some("paymentOnAccount1.text") => "Late payment interest for payment on account 1 of 2"
-    case Some("paymentOnAccount2.text") => "Late payment interest for payment on account 2 of 2"
-    case Some("balancingCharge.text") => "Late payment interest for remaining balance"
-    case _ => s"Some unexpected message key: $key"
+  private def getAllocationDescriptionFromKey(key: Option[String]): String = {
+    key match {
+      case Some("chargeSummary.paymentAllocations.mfaDebit") => "Payment put towards HMRC adjustment"
+      case Some("chargeSummary.paymentAllocations.poa1.incomeTax") => "Income Tax for payment on account 1 of 2"
+      case Some("chargeSummary.paymentAllocations.poa1.nic4") => "Class 4 National Insurance for payment on account 1 of 2"
+      case Some("chargeSummary.paymentAllocations.poa2.incomeTax") => "Income Tax for payment on account 2 of 2"
+      case Some("chargeSummary.paymentAllocations.poa2.nic4") => "Class 4 National Insurance for payment on account 2 of 2"
+      case Some("chargeSummary.paymentAllocations.bcd.incomeTax") => "Income Tax for remaining balance"
+      case Some("chargeSummary.paymentAllocations.bcd.nic2") => "Class 2 National Insurance for remaining balance"
+      case Some("chargeSummary.paymentAllocations.bcd.vcnic2") => "Voluntary Class 2 National Insurance for remaining balance"
+      case Some("chargeSummary.paymentAllocations.bcd.nic4") => "Class 4 National Insurance for remaining balance"
+      case Some("chargeSummary.paymentAllocations.bcd.sl") => "Student Loans for remaining balance"
+      case Some("chargeSummary.paymentAllocations.bcd.cgt") => "Capital Gains Tax for remaining balance"
+      case Some("paymentOnAccount1.text") => "Late payment interest for payment on account 1 of 2"
+      case Some("paymentOnAccount2.text") => "Late payment interest for payment on account 2 of 2"
+      case Some("balancingCharge.text") => "Late payment interest for remaining balance"
+      case Some("codingOut.cancelled") => "Cancelled PAYE Self Assessment (through your PAYE tax code)"
+      case Some("codingOut.accepted") =>
+        s"Amount collected through your PAYE tax code for ${dateService.getCurrentTaxYearEnd} to ${dateService.getCurrentTaxYearEnd + 1} tax year"
```

Comment:
once again, are you sure we expected to use dateService in order to extract taxYear ???

Reply:
Removed dateService